### PR TITLE
[Heartbeat] Fix Timerqueue initialization / maintainability

### PR DESCRIPTION
**Pull request by andrewvc**
Fixes https://github.com/elastic/beats/issues/26205

This PR changes our strategy from resetting timers to using a ticker with 250ms resolution. For whatever reason, reset seems to now be unstable (maybe a go runtime change?) on windows.

This PR also switches some hard to read parts of the code to using the `time.Until` syntax. I'm including this in this change to make the code easier to read in the future, which will help in future debugging. It also restructures some confusing code to make it easier to read.

I do suspect that there was a change in the go runtime implementation that caused this issue to service now since it's been around for a year and 1/2.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

`go test -timeout 30s -run '^TestQueueRunsInOrder$' github.com/elastic/beats/v7/heartbeat/scheduler/timerqueue -count=8000` should reproduce this locally on master. It should never happen on this branch

### CODE REVIEW

File: heartbeat/scheduler/timerqueue/queue.go [hunk 1]
```diff
@@ -103,16 +114,22 @@ func (tq *TimerQueue) Start() {
...
-		// Stop and drain the timer prior to reset per https://golang.org/pkg/time/#Timer.Reset
-		// Only drain if nextRunAt is set, otherwise the timer channel has already been stopped the
-		// channel is empty (and thus would block)
-		if tq.nextRunAt != nil && !tq.timer.Stop() {
-			<-tq.timer.C
-		}
-		tq.timer.Reset(tt.runAt.Sub(time.Now()))
-
+	if tq.nextRunAt == nil {
+		tq.timer.Reset(time.Until(tt.runAt))
```
Comment (andrewkroh) [inline heartbeat/scheduler/timerqueue/queue.go:118]: Is this another instance where it is resetting the timer after stopping it (timer fires, heap is emptied, timer is stopped, new task comes in)? I guess this and the other `Reset` should be replaced too.

**Review by andrewkroh — APPROVED** [review]
LGTM under the assumption the problem is with the timer Reset. It's definitely are hard one to track down. 👍

**Comment by andrewvc** [comment]
Scheduler benchmarks

This PR:
```
andrewvc@LAPTOP-80O11FM2 ~/p/b/heartbeat (fix-tq)> /home/andrewvc/.gvm/versions/go1.16.4.linux.amd64/bin/go test -benchmem -bench '^(BenchmarkScheduler)$' github.com/elastic/beats/v7/heartbeat/scheduler -benchtime=20s
goos: linux
goarch: amd64
pkg: github.com/elastic/beats/v7/heartbeat/scheduler
cpu: Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
BenchmarkScheduler-10            9716433              2692 ns/op             493 B/op         14 allocs/op
PASS
ok      github.com/elastic/beats/v7/heartbeat/scheduler 28.680s
```

Master

```
andrewvc@LAPTOP-80O11FM2 ~/p/b/heartbeat (master)> /home/andrewvc/.gvm/versions/go1.16.4.linux.amd64/bin/go test -benchmem -bench '^(BenchmarkScheduler)$' github.com/elastic/beats/v7/heartbeat/scheduler -benchtime=20s
goos: linux
goarch: amd64
pkg: github.com/elastic/beats/v7/heartbeat/scheduler
cpu: Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
BenchmarkScheduler-10            7515495              2999 ns/op             504 B/op         13 allocs/op
PASS
ok      github.com/elastic/beats/v7/heartbeat/scheduler 44.524s
```

**Comment by andrewvc** [comment]
Closing in favor of https://github.com/elastic/beats/pull/27006

**Pull request closed without merging**